addons/docs/src/lib/docgen/createPropDef.ts

```diff
@@ -4,31 +4,45 @@ import { TypeSystem, DocgenInfo, DocgenType } from './types';
 import { JsDocParsingResult } from '../jsdocParser';
 import { createDefaultValue } from './createDefaultValue';
 
-export type PropDefFactory = (
-  propName: string,
-  docgenInfo: DocgenInfo,
-  jsDocParsingResult?: JsDocParsingResult
-) => PropDef;
+export interface PropDefFactoryProps {
+  name: string;
```

Comment:
I do prefer `propName`, `name` doesn't give enough context in this case.

Reply:
I am not sure we need this interface.

I feel like it's only adding complexity without much gain. If you want to prevent redefining the argument for every factories maybe you could use rest + spread operators?